platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/reconnect/VirtualTeacherTreeView.java

```diff
@@ -188,6 +251,11 @@ public void registerResponseForNode(final Long node, final boolean learnerHasNod
                 ? ConcurrentNodeStatusTracker.Status.KNOWN
                 : ConcurrentNodeStatusTracker.Status.NOT_KNOWN;
         nodeStatusTracker.set(node, status);
+        if (node == lastNodeAwaitingReporting) {
+            synchronized (node) {
+                node.notify();
```

Comment:
Shouldn't it be `notifyAll`?

Reply:
Currently, the code calling wait() on this object is single-threaded, so technically the notify() should suffice. However, I replaced it with a call to notifyAll() in the latest commit in case future implementations would make use of multiple threads.